src/trust_anchor.rs

```diff
@@ -4,11 +4,29 @@ use crate::cert::{lenient_certificate_serial_number, Cert};
 use crate::der;
 use crate::error::{DerTypeId, Error};
 
-/// Interprets the given DER-encoded certificate as a `TrustAnchor`. The
-/// certificate is not validated. In particular, there is no check that the
-/// certificate is self-signed or even that the certificate has the cA basic
-/// constraint.
-pub fn extract_trust_anchor<'a>(cert: &'a CertificateDer<'a>) -> Result<TrustAnchor<'a>, Error> {
+/// Interprets the given pre-validated DER-encoded certificate as a `TrustAnchor`.
+///
+/// This function extracts the components of a trust anchor (see [RFC 5280 6.1.1]) from
+/// an X.509 certificate obtained from a source trusted to have appropriately validated
+/// the subject name, public key, and name constraints in the certificate. E.g. your
+/// operating system's trust store.
```

Comment:
Nit: at some point I picked up the advice not to use "i.e." or "e.g." and instead write out something like "for example". I think that could apply here, I would probably also not start a new sentence for it.

Reply:
That seems like reasonable advice :+1: Fixed up.